# Assignment 02: make a Python notebook to analyze CORGIS data

### Digital Humanities 140 | Jane Huang

## Introduction: The Dataset

This is the Broadway Python Library from the CORGIS Dataset Project. This dataset is about Broadway shows, which are grouped over weeklong periods. The dataset is made available by the Broadway League. The dataset is obtained from the [CORGIS website](https://corgis-edu.github.io/corgis/python/broadway/).

First, import the broadway data file:

In [4]:
# import Broadway dataset
import broadway
production = broadway.get_production()

Read the broadway data file:

In [6]:
type(production)

list

- The type of the broadway data file is a list.

In [7]:
len(production)

31296

- The length of the broadway data file is 31296.

## Exploration

Let's explore some of the variables in the dataset.

Using for loop:

In [58]:
# print Broadway productions each week with name and attendance
for i in production[5010:5030]:
    print('On '+ str(i['Date']['Full']) + ', ' + str(i['Show']['Name']) + ' was performed in ' + str(i['Show']['Theatre']))

On 4/25/1999, The Lonesome West was performed in Lyceum
On 4/25/1999, The Phantom Of The Opera was performed in Majestic
On 4/25/1999, The Scarlet Pimpernel was performed in Minskoff
On 4/25/1999, The Sound Of Music was performed in Martin Beck
On 4/25/1999, The Weir was performed in Walter Kerr
On 4/25/1999, Via Dolorosa was performed in Booth
On 4/25/1999, You'Re A Good Man, Charlie Brown was performed in Ambassador
On 5/2/1999, Amy'S View was performed in Ethel Barrymore
On 5/2/1999, Annie Get Your Gun was performed in Marquis
On 5/2/1999, Art was performed in Royale
On 5/2/1999, Beauty And The Beast was performed in Palace
On 5/2/1999, Cabaret 98 was performed in Studio 54 ('98)
On 5/2/1999, Cats was performed in Winter Garden
On 5/2/1999, Chicago was performed in Shubert
On 5/2/1999, Closer was performed in Music Box
On 5/2/1999, Death Of A Salesman 99 was performed in Eugene O'Neill
On 5/2/1999, Footloose was performed in Richard Rodgers
On 5/2/1999, Fosse was performed in Broadh

Using mathematical operation, conditional expression, and for loop (for demonstration, print only the first 20 indices in the dataset):

In [59]:
# for the first 20 indices in the broadway dataset
for i in production[0:20]:
    # if performance is not 0 for the week
    if i['Statistics']['Performances'] != 0:
        # print the average attendance
        print('For ' + str(i['Date']['Full']) + ', the average number of people who attended performances over the week was ' + str(i['Statistics']['Attendance']/i['Statistics']['Performances']) + '.')
    # else if performance is 0
    else:
        # print 'there is no performance'
        print('For ' + str(i['Date']['Full']) + ', there was no performance for the week.')

For 8/26/1990, the average number of people who attended performances over the week was 687.5.
For 3/24/1991, there was no performance for the week.
For 3/31/1991, there was no performance for the week.
For 4/7/1991, there was no performance for the week.
For 4/14/1991, the average number of people who attended performances over the week was 2743.25.
For 4/21/1991, the average number of people who attended performances over the week was 1759.5.
For 4/28/1991, the average number of people who attended performances over the week was 1758.125.
For 5/5/1991, the average number of people who attended performances over the week was 1758.0.
For 5/12/1991, the average number of people who attended performances over the week was 1737.0.
For 5/19/1991, the average number of people who attended performances over the week was 1717.25.
For 5/26/1991, the average number of people who attended performances over the week was 1737.125.
For 6/2/1991, the average number of people who attended performance

Defining function & using mathematical operation:

The gross potential is the maximum amount an engagement can possibly earn based on calculations involving ticket prices, seating capacity, and the number of performances.

In [49]:
# define function that calculates the gross potential for a production
def gp(gross, grossPotentialPercentage):
    if grossPotentialPercentage != 0:
        return gross/grossPotentialPercentage
    else:
        return 'Gross potential not available'

In [62]:
# use the newly defined function gp()
gp(production[0]['Statistics']['Gross'], production[0]['Statistics']['Gross Potential'])

'Gross potential not available'

- The gross potential was not available.

In [63]:
# use the newly defined function gp()
gp(production[10000]['Statistics']['Gross'], production[10000]['Statistics']['Gross Potential'])

10384.91

- The gross potential was 10384.91.